src/Dapr.Client/DaprClient.cs

```diff
@@ -15,6 +17,50 @@ namespace Dapr.Client
     /// </summary>
     public abstract class DaprClient
     {
+        /// <summary>
+        /// <para>
+        /// Creates an <see cref="HttpClient" /> that can be used to perform Dapr service
+        /// invocation using <see cref="HttpRequestMessage" /> objects.
+        /// </para>
+        /// <para>
+        /// The client will read the <see cref="HttpRequestMessage.RequestUri" /> property, and 
+        /// interpret the hostname as the destination <c>app-id</c>. The <see cref="HttpRequestMessage.RequestUri" /> 
+        /// property will be replaced with a new URI with the authority section replaced by <paramref name="daprEndpoint" />
+        /// and the path portion of the URI rewitten to follow the format of a Dapr service invocation request.
+        /// </para>
+        /// </summary>
+        /// <param name="appId">
+        /// An optional <c>app-id</c>. If specified, the <c>app-id</c> will be configured as the value of 
+        /// <see cref="HttpClient.BaseAddress" /> so that relative URIs can be used.
+        /// </param>
+        /// <param name="daprEndpoint">The HTTP endpoint of the Dapr process to use for service invocation calls.</param>
+        /// <returns>An <see cref="HttpClient" /> that can be used to perform service invocation requests.</returns>
+        /// <remarks>
+        /// <para>
+        /// The <see cref="HttpClient" /> object is intended to be a long-lived and holds access to networking resources.
+        /// Since the value of <paramref name="daprEndpoint" /> will not change during the lifespan of the application,
+        /// a single client object can be reused for the life of the application.
+        /// </para>
+        /// </remarks>
+        public static HttpClient CreateInvokeClient(string appId = null, string daprEndpoint = null)
```

Comment:
Why are we not using the builder pattern here like for the grpc case?

Reply:
Builder APIs are complex and undiscoverable. I'd prefer not to use a sledgehammer to squash a fly. 

This is a convenience API that does glue code for something the user already can do (create an HttpClient). If it becomes complex then there's no value.